greymatter/inputs.cue

```diff
@@ -1,62 +1,94 @@
 package greymatter
 
-config: {
-	// enable Spire-based mTLS
-	spire: bool | *false @tag(spire,type=bool)
-}
-
+// WARNING - Configurations in the mesh section must not be changed
+// arbitrarily because a tenant's mesh configurations hook into an existing
+// enterprise's core mesh configurations. Change these values in accordance
+// with enterprise-level configurations. This may require coordination with
+// an enterprise operations team.
 mesh: {
-	metadata: {
-		name: string | *"greymatter-mesh"
-	}
-	spec: {
-		zone: string | *"default-zone"
-		images: {
-			proxy: string | *"quay.io/greymatterio/gm-proxy:1.7.0"
-		}
+	// Mesh zone must match the enterprise zone so services are discovered
+	// by the greymatter control plane. Consult with the enterprise operations
+	// team if you are not sure about this value. It's important that the zone
+	// matches the enterprise zone before deploying tenant services.
+	zone: "default-zone"
+	// Mesh name must match the enterprise mesh name so services appear
+	// in the greymatter dashboard.
+	name: "greymatter-mesh"
+	// Enabling Spire creates the necessary configurations to support sidecar
+	// to sidecar mutual TLS (mTLS) connections. The pre-requisite for this
+	// feature is that the enterprise team has deployed Spire or is using an
+	// external one. Therefore, enabling Spire on tenant sidecars should be
+	// done in accordance with enterprise-level Spire settings. The above mesh
+	// and zone names are used in Spire configuration and must match enterprise
+	// configurations.
+	enable_spire: false
+	// Sidecars publish metrics to Redis in support of sidecar/service
+	// health checks in the dashboard. A tenant does not need to change
+	// these configurations. Doing so may prevent operators from using
+	// health checks to troubleshoot sidecars/services.
+	redis: {
+		// The Redis key identifies the target for sidecar to Redis
+		// egress communication.
+		key: "redis"
+		// The Redis port indicates the localhost port from which egress
+		// requests flow out from the sidecar to Redis.
+		ingress_port: 10910
 	}
 }
 
-defaults: {
-	redis_cluster_name: "redis"
-
-	ports: {
-		default_ingress: 10808
-		edge_ingress:    10809
-		redis_ingress:   10910
-		metrics:         8081
-	}
+// A subset of a tenant's sidecar configurations can be modified in
+// isolation from the enterprise. These do not impact global sidecar
+// configurations.
+sidecars: {
```

Comment:
Maybe not call this block sidecar? I'm open to ideas but a sidecar may imply like vector instances, or other things running in the pod. Maybe make this greymatter specific?

Reply:
I also think we can get rid of this block too. Is there a good case for default ingress ports to change? I can't see one.

Reply:
The only case is non-k8s deployments this will need to be modified. I'm wondering if we be clever about the names where we have a `k8s` block with `sidecar-ingress` and then non-k8s deployments can have a block like `eks: ingress-port`.

Reply:
@alecholmez I'm not sure I follow what you're saying. Isn't sidecar ingress port ubiquitous across various deployment platforms, k8s and non-k8s?

Reply:
Unfortunately no I ran into this when deploying this CUE with an E2E mesh. Every port needs to be unique otherwise you'll get collisions. Maybe thats a one off thing we can document I guess I'm trying to thinking about multiple deployment scenarios instead of tailoring to k8s. Though in reality we're targeting k8s so this may be a point we punt down the line

Reply:
Why then do things not break when we deploy core services? They all use 10808 as the ingress port.

Reply:
As long as non-k8s deployments have a 1-to-1 mapping between machines and sidecar proxies then there won't be port collisions. In the case of gm-e2e, the entire mesh runs on one virtual machine, hence the port conflicts.

Reply:
Ahh, I get the port conflict issue now.